scripts/create-secrets/k8sExecution.js

```diff
@@ -0,0 +1,320 @@
+];
+
+let env = process.env;
+function k8sExecution(config, shouldNotAsk = false) {
+    try {
+        return doK8sExecution(config, shouldNotAsk);
+    } catch (e) {
+        return Promise.reject(e);
+    }
+}
+
+function doK8sExecution(config, shouldNotAsk = false) {
+    env = getEnvByClusterType(config);
+    let configData = Object.assign({}, config.all);
+    const ifAllowEnvVarOverride = configData["allow-env-override-settings"];
+    if (ifAllowEnvVarOverride) {
+        configData = overrideSettingWithEnvVars(configData);
+    }
+    validKubectl();
+    let p = Promise.resolve().then(function() {
+        configData["cluster-namespace"] = trim(configData["cluster-namespace"]);
+        if (!configData["cluster-namespace"]) {
+            throw new Error(
+                "Cluster namespace cannot be empty! \n " +
+                    "If you've set cluster namespace, make sure it's not overrided by env variable."
+            );
+        }
+    });
+    if (!checkNamespace(configData["cluster-namespace"])) {
+        if (shouldNotAsk) {
+            throw new Error(
+                `Namespace ${
+                    configData["cluster-namespace"]
+                } doesn't exist. Please create and try again.`
+            );
+        }
+        p = p
+            .then(
+                askIfCreateNamespace.bind(null, configData["cluster-namespace"])
+            )
+            .then(function(ifCreate) {
+                if (!ifCreate) {
+                    console.log(
+                        chalk.yellow(
+                            `You need to create namespace \`${
+                                configData["cluster-namespace"]
+                            }\` before try again.`
+                        )
+                    );
+                    process.exit();
+                } else {
+                    createNamespace(configData["cluster-namespace"]);
+                }
+            });
+    }
+    return p.then(function() {
+        const namespace = configData["cluster-namespace"];
+        if (configData["use-cloudsql-instance-credentials"] === true) {
+            createFileContentSecret(
+                namespace,
+                "cloudsql-instance-credentials",
+                "credentials.json",
+                configData["cloudsql-instance-credentials"]
+            );
+        }
+        if (configData["use-storage-account-credentials"] === true) {
+            createFileContentSecret(
+                namespace,
+                "storage-account-credentials",
+                "db-service-account-private-key.json",
+                configData["storage-account-credentials"]
+            );
+        }
+        if (configData["use-smtp-secret"] === true) {
+            createSecret(namespace, "smtp-secret", {
+                username: configData["smtp-secret-username"],
+                password: configData["smtp-secret-password"]
+            });
+        }
+        (function() {
```

Comment:
Why is this in an immediately executing function?

Reply:
Only want to create a new scope so I don't need to worry about any temporary variables I defined.
In this case, `const data`

Reply:
If it makes sense to have it in its own function, you might as well create an actual named function right? Like `function createDbPasswords(dbPasswordNames) {`

Reply:
Sounds good~ moved to a separate `createDbPasswords` function